vmdb/app/models/ems_kubernetes.rb

```diff
@@ -13,17 +13,41 @@ def self.description
 
   def self.raw_connect(hostname, port, api_version)
     require 'kubeclient'
-    require 'uri'
-
-    uri = URI::HTTP.build(:path => "/api", :port => port.to_i)
+    api_endpoint = raw_api_endpoint(hostname, port)
+    Kubeclient::Client.new(api_endpoint, api_version)
+  end
 
+  def self.raw_api_endpoint(hostname, port)
+    require 'uri'
     # URI::Generic#hostname= was added in ruby 1.9.3 and will automatically
     # wrap an ipv6 address in []
-    uri.hostname = hostname
-    Kubeclient::Client.new uri.to_s, api_version
+    URI::HTTP.build(:host => hostname, :port => port.to_i, :path => '/api')
```

Comment:
@simon3z we have build the URI object and THEN call uri.hostname = hostname or you have to deal with enclosing IPv6 addresses with [] on your own.

That is, until we upgrade to a ruby that has my PR included https://github.com/ruby/ruby/pull/765, which is currently on trunk ruby:  https://github.com/ruby/ruby/commit/4ec9793ba88b27028c826c6bc6c49ee59df8d920